### PR TITLE
Feat 8/user api model

### PR DESCRIPTION
**Pull request by sueelly**
## #️⃣ Related Issue

Linked Feature backlog: close (put issue number here with #)

## 📝 Purpose of PR

Please briefly explain your work in this PR.

## Contents

List of work items performed.

## Checklist

-   [ ] Does commit messages follow the convention?
-   [ ] Are variable names brief but descriptive?
-   [ ] Is the code clean and easy to understand?
-   [ ] If a new feature has been added, or a bug fixed, has a test been added to confirm good behavior?
-   [ ] Does the test successfully test edge and corner cases?

## 💬 Review Requirements (Optional)

Is there anything you can praise about this PR? Start with praise.
Make a review, leaving kind comments and suggesting changes where needed (to resolve the above).

### CODE REVIEW

File: snapnow-api/src/main/java/kr/flab/snapnow/user/model/dto/response/profile/UserProfileResponse.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    private String postCountText;
+    private int followerCount;
+    private String followerCountText;
+    private int followingCount;
+    private String followingCountText;
+
+    private FollowStatus followStatus;
+    // Todo: follower 알림 설정 필드 추가
+
+    public static class UserProfileResponseBuilder {
```
Comment (f-lab-jason) [inline snapnow-api/src/main/java/kr/flab/snapnow/user/model/dto/response/profile/UserProfileResponse.java:32]: `@Builder`가 있는데 따로 builder class는 왜 필요한건가요?

File: snapnow-api/src/main/java/kr/flab/snapnow/user/model/dto/response/setting/NotificationSettingResponse.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import kr.flab.snapnow.domain.user.domain.enums.setting.settingType.NotificationSettingType;
+import kr.flab.snapnow.domain.user.domain.exception.InvalidSettingException;
+
+@SuperBuilder
+public class NotificationSettingResponse
+        extends UserSettingResponse<NotificationSettingType, NotificationSettingOption> {
+
+    protected NotificationSettingResponse(SettingCategory category,
+            Map<NotificationSettingType, NotificationSettingOption> settings) {
+        if (!SettingCategory.NOTIFICATION.equals(category)) {
```
Comment (f-lab-jason) [inline snapnow-api/src/main/java/kr/flab/snapnow/user/model/dto/response/setting/NotificationSettingResponse.java:18]: 이 친구들은 생성하는데 category 정보가 왜 필요한건가요?

**Review by f-lab-jason — COMMENTED** [review]
음 솔직히 비지니스 로직없이 모델만 있으니까 용도가 잘 안보이는것 같긴 해요.
멘토링 세션때 이야기해보면 좋겠네요

File: snapnow-api/src/main/java/kr/flab/snapnow/user/model/dto/response/profile/UserProfileResponse.java (hunk 1 above)
Comment (sueelly) [reply snapnow-api/src/main/java/kr/flab/snapnow/user/model/dto/response/profile/UserProfileResponse.java:32]: builder을 커스텀하게 사용하기 위해서입니다. `@Builder`만 사용하는 경우 로직을 넣을 수가 없습니다. todo에 적은 것처럼 build()가 호출될 때 변환하여 값을 추가하는 로직을 넣어야하는데, 그때 사용됩니다.

다만 현재는 모두 builder가 호출하는 생성자에서 처리하는 방식으로 바꾸고 있기 때문에, 통일하도록 하겠습니다.

File: snapnow-api/src/main/java/kr/flab/snapnow/user/model/dto/response/setting/NotificationSettingResponse.java (hunk 2 above)
Comment (sueelly) [reply snapnow-api/src/main/java/kr/flab/snapnow/user/model/dto/response/setting/NotificationSettingResponse.java:18]: 없어도 전혀 상관없을 것 같습니다.
category, type, option을 계속 묶어서 사용하다보니 들어간 코드인 듯 한데, 삭제도록 하겠습니다
